diff --git a/nixos/modules/installer/cd-dvd/iso-image.nix b/nixos/modules/installer/cd-dvd/iso-image.nix
index 31c8516eb187..3345b090025c 100644
--- a/nixos/modules/installer/cd-dvd/iso-image.nix
+++ b/nixos/modules/installer/cd-dvd/iso-image.nix
@@ -465,6 +465,9 @@ let
             }
           fi
         ''}
+        menuentry 'EFI Shell' --class settings {
+          chainloader /EFI/BOOT/shell.efi
+        }
         menuentry 'Firmware Setup' --class settings {
           fwsetup
           clear
@@ -480,6 +483,8 @@ let
         grub-script-check $out/EFI/BOOT/grub.cfg
 
         ${refind}
+
+        cp -v ${pkgs.edk2-uefi-shell}/shell.efi $out/EFI/BOOT/
       '';
 
   efiImg =
