diff --git a/Makefile b/Makefile
index b49c57a..28bd20c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,16 +1,16 @@
-AUDIO_ROOT=$(PWD)
-
-obj-m := ipc/audio-pkt.o
+obj-m += ipc/audio-pkt.o
 obj-m += dsp/spf-core.o
 obj-m += dsp/msm_audio_mem.o
 
-EXTRA_CFLAGS += -I$(AUDIO_ROOT)/include
+EXTRA_CFLAGS += -I$(PWD)/include
+
+default: modules
 
-all:
-	$(MAKE) -C $(KERNEL_SRC) M=$(shell pwd) modules
+modules:
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) modules
 
 modules_install:
-	$(MAKE) INSTALL_MOD_STRIP=1 -C $(KERNEL_SRC) M=$(shell pwd) modules_install
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) modules_install
 
 clean:
 	find ./ -iname "*.o" -delete
