diff --git a/Makefile b/Makefile
index e2b57ed..1fada67 100644
--- a/Makefile
+++ b/Makefile
@@ -92,15 +92,11 @@ DTBS := \
 
 all: $(OUT_DIR)/sltest.efi $(OUT_DIR)/slbounce.efi $(OUT_DIR)/dtbhack.efi
 
-.INTERMEDIATE: $(GNUEFI_DIR)/inc/elf.h
-$(GNUEFI_DIR)/inc/elf.h:
-	ln -sf /usr/include/elf.h $(GNUEFI_DIR)/inc/elf.h
-
-$(LIBEFI_A): $(GNUEFI_DIR)/inc/elf.h
+$(LIBEFI_A):
 	@echo [ DEP ] $@
 	@$(MAKE) -C$(GNUEFI_DIR) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH) lib
 
-$(LIBGNUEFI_A): $(GNUEFI_DIR)/inc/elf.h
+$(LIBGNUEFI_A):
 	@echo [ DEP ] $@
 	@$(MAKE) -C$(GNUEFI_DIR) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH) gnuefi
 
diff --git a/README.md b/README.md
index 9b7e526..53a4b49 100644
--- a/README.md
+++ b/README.md
@@ -44,7 +44,7 @@ Make sure `tcblaunch.exe` is placed in the root of the FS, then load the EFI dri
 fs0:\> load slbounce.efi
 ```
 
-The driver will replace `BS->ExitBootServices` in the system table with it's own
+The driver will replace `BS->ExitBootServices` in the system table with its own
 function. It will call EBS and perform switch to EL2 right after. Thus your bootloader
 (i.e. grub or linux's efi-stub) would experence the cpu swithing to EL2 when it calls
 EBS.
